install.sh

```diff
@@ -0,0 +1,211 @@
+#!/bin/bash
+
+set -u
+
+DOTENV_LINTER="dotenv-linter"
+DOTENV_LINTER_REPO="${DOTENV_LINTER}/${DOTENV_LINTER}"
+DOTENV_LINTER_GITHUB="https://github.com/${DOTENV_LINTER_REPO}"
+DOTENV_LINTER_RELEASE_API="https://api.github.com/repos/${DOTENV_LINTER_REPO}/releases/latest"
+
+main() {
+    require_cmd uname
+    require_cmd mkdir
+    require_cmd grep
+    require_cmd cut
+    require_cmd rm
+
+    local _target_version
+    if [ $# -eq 0 ]; then
+        get_latest_version || return 1
+        _target_version="$RETVAL"
+    else
+        _target_version="$1"
+    fi
+
+    get_architecture || return 1
+    local _arch="$RETVAL"
+
+    local _ext
+    case $_arch in
+        win-*) _ext=".zip" ;;
+        *) _ext=".tar.gz" ;;
+    esac
+
+    local _archive_name="${DOTENV_LINTER}-${_arch}${_ext}"
+    local _url="${DOTENV_LINTER_GITHUB}/releases/download/${_target_version}/${_archive_name}"
+
+    # Installation
+    local _dir="$HOME/.${DOTENV_LINTER}"
+    local _version_dir="${_dir}/${_target_version}"
+
+    if [ ! -e "$_version_dir/${DOTENV_LINTER}" ]; then
+        mkdir -p "$_version_dir"
+
+        local _archive_path="${_version_dir}/${_archive_name}"
+        download "${_url}" "${_archive_path}" || return 1
+
+        case $_archive_path in
+            *.zip)
+                required_cmd unzip
+                unzip "${_archive_path}" -d "${_version_dir}"
+            ;;
+            *)  require_cmd tar
+                tar -xzf "${_archive_path}" -C "${_version_dir}"
+            ;;
+        esac
+
+        rm "${_archive_path}"
+
+        println "Successful download \"dotenv-linter\" to ${_version_dir}"!
+    else
+        println "WARN:: ${_version_dir} already exists! We just switch the version to ${_target_version}"
+        println "Or you can manually remove dir: 'rm -rf ${_version_dir}' and reinstall."
+    fi
+
+    # Setup installed version as executable
+    setup_executable "${_dir}" "${_target_version}" || return 1
+
+    return 0;
+}
+
+# $1 - dotenv-linter dir, $2 - target version
+setup_executable() {
+    local _bin_dir="$1/bin"
+    if [ ! -d "$_bin_dir" ]; then
+	    mkdir -p "$_bin_dir"
+    fi
+
+    local _bin_file="${_bin_dir}/${DOTENV_LINTER}"
+    ln -sf "$1/$2/${DOTENV_LINTER}" "${_bin_file}"
```

Comment:
Why do we do that?

Reply:
I was thinking about supporting multi-versioning, but that seems overkill.
Symbolic links are used by many programs for local use without the need to replace the entire versioned package.

We will change this, because I will adhere to the approach mentioned above.